### PR TITLE
Fix cpu & memory change in content library

### PR DESCRIPTION
**Pull request by ansilh**
Fixes https://github.com/rancher/rancher/issues/32761 
The VM gets deployed from the content library was taking defaults from
OVF template instead of driver input. Added code to reconfigure VM
based on the input from the driver.

### CODE REVIEW

File: drivers/vmwarevsphere/create.go [hunk 1]
```diff
@@ -362,7 +362,23 @@ func (d *Driver) createFromLibraryName() error {
...
+	// Reconfiguration of the VM based on driver inputs follows
+
 	vm := obj.(*object.VirtualMachine)
+	spec := types.VirtualMachineConfigSpec{
+		NumCPUs:    int32(d.CPU),
+		MemoryMB:   int64(d.Memory),
+		VAppConfig: d.getVAppConfig(),
+        }
+
+	task, err := vm.Reconfigure(d.getCtx(), spec)
```
Comment (leodotcloud) [inline drivers/vmwarevsphere/create.go:375]: Don't we need an err handler here?

Comment (ansilh) [reply drivers/vmwarevsphere/create.go:375]: Added error handling

File: drivers/vmwarevsphere/create.go [hunk 2]
```diff
@@ -362,7 +362,23 @@ func (d *Driver) createFromLibraryName() error {
...
 
+	// At this point, the VM is deployed from content library with defaults from template
+	// Reconfiguration of the VM based on driver inputs follows
+
 	vm := obj.(*object.VirtualMachine)
+	spec := types.VirtualMachineConfigSpec{
+		NumCPUs:    int32(d.CPU),
+		MemoryMB:   int64(d.Memory),
+		VAppConfig: d.getVAppConfig(),
+        }
```
Comment (luthermonson) [inline drivers/vmwarevsphere/create.go:373]: go fmt

Comment (ansilh) [reply drivers/vmwarevsphere/create.go]: Done

**Comment by luthermonson** [comment]
thanks for the pr! @ansilh

**Pull request merged**